bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/sequence/entity/pcm/AbstractPCMActionSequenceElement.java

```diff
@@ -0,0 +1,46 @@
+package org.palladiosimulator.dataflow.confidentiality.analysis.sequence.entity.pcm;
+
+import java.util.Objects;
+import java.util.Stack;
+
+import org.eclipse.emf.ecore.EObject;
+import org.palladiosimulator.dataflow.confidentiality.analysis.sequence.entity.AbstractActionSequenceElement;
+import org.palladiosimulator.pcm.core.composition.AssemblyContext;
+
+public abstract class AbstractPCMActionSequenceElement<T extends EObject> extends AbstractActionSequenceElement<T> {
+
+    private final Stack<AssemblyContext> context;
```

Comment:
Might be better to use somethin like a [Deque](https://docs.oracle.com/en/java/javase/17/docs/api//java.base/java/util/Deque.html) since this is the recommende new data structure